templates/types/streaming/nextjs/app/components/ui/chat/chat-message/chat-sources.tsx

```diff
@@ -1,123 +1,193 @@
-import { Check, Copy } from "lucide-react";
+import { Check, Copy, FileText } from "lucide-react";
+import Image from "next/image";
 import { useMemo } from "react";
 import { Button } from "../../button";
+import { FileIcon } from "../../document-preview";
 import {
   HoverCard,
   HoverCardContent,
   HoverCardTrigger,
 } from "../../hover-card";
 import { useCopyToClipboard } from "../hooks/use-copy-to-clipboard";
-import { SourceData } from "../index";
+import { DocumentFileType, SourceData, SourceNode } from "../index";
 import PdfDialog from "../widgets/PdfDialog";
 
-const SCORE_THRESHOLD = 0.3;
+const SCORE_THRESHOLD = 0.25;
 
-function SourceNumberButton({ index }: { index: number }) {
-  return (
-    <div className="text-xs w-5 h-5 rounded-full bg-gray-100 mb-2 flex items-center justify-center hover:text-white hover:bg-primary hover:cursor-pointer">
-      {index + 1}
-    </div>
-  );
-}
-
-type NodeInfo = {
-  id: string;
-  url?: string;
+type Document = {
+  url: string;
+  sources: SourceNode[];
 };
 
 export function ChatSources({ data }: { data: SourceData }) {
-  const sources: NodeInfo[] = useMemo(() => {
-    // aggregate nodes by url or file_path (get the highest one by score)
-    const nodesByPath: { [path: string]: NodeInfo } = {};
-
+  const documents: Document[] = useMemo(() => {
+    // group nodes by document (a document must have a URL)
+    const nodesByUrl: Record<string, SourceNode[]> = {};
     data.nodes
       .filter((node) => (node.score ?? 1) > SCORE_THRESHOLD)
```

Comment:
@thucpn we need this filter, otherwise we will always show nodes even if the data is unrelated